[5.09.29] Big (petro)chemistry update

JohannesGaessler opened the pull request:
I'm currently working on an update to GregTech's chemistry.
I'm focusing mostly on synthetic materials.
So far I've added/changed the following things:

===API Changes===
*Added a MaterialBuilder class for convenience. You only have to set the values relevant for your material, the rest will be set to a default value.
*Expanded the material system so you can automatically generate gases or fluids from your material
*Distillery Recipes can now have up to one solid output
*Chemical recipes can now have up to two solid outputs
*Added convenience methods for creating Fluid Pipes/Item Pipes
*Added covenience methods for getting frequently required Material components (example: Materials.Sulfur.getDust(amount))
*Added a convenience method for getting an Integrated circuit (GT_Utility.getIntegratedCircuit(config))

===Gameplay Changes===
*There are huge changes made to the chemistry of GT5U with the goal to make it more realistic. 
*Instead of nebulous compound substances like Naphtha you now generally have well-defined chemicals which are reacting according to real chemical formulas.
*Chemical recipes are now stoichiometrically correct, the fluid amount is interpreted as the number of molecules / atoms. For example you now have 2 H + 1 O -> 1 H2O instead of 2 H + 1 O -> 3 H2O. Electrolyzation and Centrifugation recipes have '''not''' been changed since that would completely break the game's balance.
*The recipes for Rocket Fuel and for most plastics and acids have been changed, typically with the effect that their production now requires more machines, but also yields a lot more material. Old recipes can be reenabled via Config.
*A synthetic recipe for Glue has been added
*A new plastic has been added: Polyvinyl Chloride, also know as PVC. Currently has no use except for making item pipes.
*Added Polytetrafluoroethylene Fluid Pipes.
*Charcoal Production now yields a great variety of fluids and gases as a Byproduct
*Added Fermented Biomass. Can be obtained from fermenting IC2 Biomass or Forestry Biomass. Can be distilled for fuel (mostly Methane) or chemicals. To run one LV Turbine you need one LV Brewery, four LV Fermenters, and one LV Distillery. Without a Distillation Tower about ~50% of the EU produced is spent in producing the Methane.
*Ashes no longer yield Carbon Dust. They now yield oxidized Minerals.
*Stone Dust can now be centrifuged. It yields minerals so that the netto amount of elements matches up with the mass percentage of elements found in earth's continental crust.
*Added a new Multiblock: The Large Chemical Reactor. Mostly has the same function as the Chemical Reactor. Can accept several fluids instead of one fluid + cells. Does not lose efficiency when overclocked. Mandatory for some recipes with a lot of inputs.
*A full description of the new chemical production processes can be found at http://ftb.gamepedia.com/Chemistry_(GregTech_5_Unofficial)


Compiled version of the branch: https://mega.nz/#!tnQTRC5b!WOP0UkgkDozuO3Xh3pumvsVTA-JRftexzv7ub0f0Sdw

Commit:
Added a lot of (petro)chemistry stuff, expanded the API

Created an adapter class for creating materials.
Materials now have bit flags for automatic gas/fluid creation.
Added a new convenience method for adding distillery recipes.
Recipes now show the exact recipe duration instead of the duration
rounded down to the nearest int.

Added new Materials with relevant recipes:
Acetic Acid, Acetone, Calcium Acetate, Charcoal Byproducts, Carbon
Monoxide, Ethanol, Ethylene, Methanol, Methyl Acetate, Polyvinyl
Acetate, Propene, Sulfuric Ethylene, Vinyl Acetate

Made vinegar visible.

Commit:
Fixed a typo, Added a missing glue recipe

OvermindDL1 commented:
I prefer these changes, but I could also see the changes being behind an option that optionally disables the changes to go back to the old method, however that might be too much work to care about, but would be useful for existing worlds...

Techlone commented on src/main/java/gregtech/api/enums/MaterialAdapter.java:
```diff
@@ -0,0 +1,219 @@
+
+import gregtech.api.objects.MaterialStack;
+
+public class MaterialAdapter {
```
Maybe 'MaterialBuilder' will be better to name this class. Why an adapter?

leagris commented:
You could also add:
Butane
Base product for Butadiene
Base product for PolyButadiene synthetic rubber additive or substitute.

Styrene
Base product for Polystyrene

If you add acrylics 
Combine Acrylic, Butadiene, Nitrogen and Styrene for strong ABS Plastic. (Acrylonitrile Butadiene Styrene)

leagris commented:
Wood alcohol is Methanol.
Methanol is result of fermenting lignite that's part of strong plant structures including wood.
Methanol an interesting chemical on its own.
Lignite is another very interesting one also.
Can be used as filler for rubbers for some bakelite or compressed wood chips composite planks or plywood. Lignite is a natural glue of cellulose fibers in wood.

Commit:
Refactored MaterialAdapter to MaterialBuilder to better represent the
class functionality.

JohannesGaessler commented:
>config

I think it shouldn't be very hard to add a config for simplified chemistry that reenables old recipes for plastic; Considering it.

>MaterialAdapter

Definitely the wrong class name; When I created the class my abstract goal was to convert the existing Materials interface into one that's more usable. Then when I decided to use a builder I forgot to rename the file.

>New material suggestions

Thank you for the input, I'll read up on those.

JohannesGaessler commented:
By the way, apart from plastics I also want to change how ashes work.
Because in reality they contain anything but carbon since that is what you're burning; Carbon dust is basically coal dust.
I plan to make ashes yield non burnable minerals like Calcium, Magnesium, or Potassium.
Coal ashes will be centrifuged into regular ashes with heavy metal byproducts (Uranium, Thorium).

draknyte1 commented:
Ash changes like that are unbalanced as hell..

JohannesGaessler commented:
I'd say the balance would strongly depend on how much you actually get out of your ashes.
Regular ashes would be ~50% calcium which is not that hard to get anyways.
Coal ashes would contain maybe half a tiny pile of Thorium each and maybe a quarter tiny pile of Uranium.
And when I say coal ashes I'm not including ashes from charcoal, only fossil coal.

leagris commented:
If acrylics were available, it could allow:
- Unbreakable glass blocks, slabs, stairs and panes (clear and dyed), all borderless texture
- Merge with chemical dyes
- Possibly strong CyanoAcrylate glue

Well, organic chemistry is so huge… plenty of stuffs.
Thermo-plastics:
- [ ] Acrylic poly(methyl methacrylate) (PMMA)
- [ ] Acrylonitrile butadiene styrene (ABS)
- [ ] Polyamides (Nylon, Dacron...) use for fibers, machine parts, gears, rods, seals
- [ ] Polylactic acid (polylactide) or PLA made from plants, seed-oil
- [ ] Polybenzimidazole(PBI,short for Poly-[2,2’-(m-phenylen)-5,5’-bisbenzimidazole])
- [ ] Vinyl or poly(methyl methacrylate) (PMMA)
- [ ] Polycarbonate (PC)
- [ ] Polyether sulfone (PES)
- [ ] Poly ether ether ketone (PEEK)
- [ ] Polyetherimide (PEI) for electrical and electronic parts
- [x] Polyethylene (polyethene, polythene, PE) we already have, but missing HDPE/LDPE classes
- [ ] Polyphenylene sulfide (PPS): cookware, bearings, and pump parts for service in various corrosive environments
- [ ] Polypropylene (PP)
- [ ] Polystyrene: mentioned earlier
- [ ] Polyvinyl chloride (PVC): expand on Vinyls
- [x] Polytetrafluoroethylene (PTFE) already part of Gregtech
- [ ] Polyurethanes (PU) 

See also: https://en.wikipedia.org/wiki/List_of_synthetic_polymers

I'd say, to make it worth, it would bring plenty of possible blocks and stuffs for building, creating in Minecraft. Think (building blocks, doors, windows, furnitures, tool handles, seals, machine parts. Would need to be cheap once you have a processing line for it though.

Commit:
Added recipes for Epichlorohydrin

draknyte1 commented:
> Coal ashes would contain maybe half a tiny pile of Thorium each and maybe a quarter tiny pile of Uranium.
And when I say coal ashes I'm not including ashes from charcoal, only fossil coal.

A coal vein contains lots of coal to obtain ash from. 
Tiny piles add up quickly, on such a scale.

JohannesGaessler commented:
Let's do the math real quick:
Currently the only ways to burn coal into ash are Coal Boilers and the Bronze Blast Furnace.
These would yield 1/9 Coal Ash or less.
Then, by centrifuging you would get 1/9 * 1/9 = 1/81 Thorium and 1/9 * 1/9 * 1/2 = 1/162 Uranium.

By comparision Pitchblende, which wouldn't be all that hard to find, yields 1/5 Thorium and 1/5 Uranium (3/5 if you add Aluminium).
So even if coal veins are relatively common I'd say it's not a particularly good way of acquiring Uranium.

Bogdan-G commented:
Only liquids and their distillation?
Next Ideas:
1) Add or copy something from gt6, balancing
2) Copy the mechanics of mods to chemistry where each block is an element more fully, but you need an adjustment for the gameplay so-called "easy game" and "hard game".

1 and 2 - we can also just copy copies from real life, but also need a rebalance. There are a number of reasons why for example PVC is used, in the game with gregtech it is not so relevant for the player, why? This is one of the shortcomings of the mechanics themselves. We also (as an variant) can turnoff/change the basic vanilla mechanics even more hardy. For example, if it is necessary to turn off the infinity water and cobblogen generators. Another variant: make more actuating the extraction of resources by smaller automated production methods.

PS Someone from the players suggested even to enter if there is a heterogeneous rock composition from the depth of different strength requirements that would not allow the player to dig deeper, for example introducing far more levels for mining than vanilla. But this idea may go beyond this PR.

JohannesGaessler commented:
@Bogdan-G 
To be honest I'm not entirely sure what you're trying to tell me.

I have not looked very hard at GT6.

My current strategy for adding things is to look at the actual, real life manufacturing process of Materials that are already in the game and try to adapt those processes.
My core principle for balancing is for synthetic materials to require a lot of machines, but once you have everything set up you're able to mass produce them relatively easily.

I'm reluctant to add a new Material like PVC because there are already a lot of plastics in the game that are not being used.
Maybe I'll look into areas where you can replace metals with plastics.
For instance I could add more plastic fluid pipe and some plastic item pipes.

Long term I'd also like to try and add a Large Chemical Reactor (definitely not part of this update).
I imagine it as a multiblock that can take multiple fluids for chemical recipes instead of one fluid and one or two fluid cells.
I plan to add something along the lines of Corrosion Resistant Machine Casing for that; If I remember correctly Polytetraflourethylene is corrosion resistant so I'll make that part of the recipe.

I will very likely add a config that let's you reenable the old recipes.

Commit:
Enabled solid output for Distilleries, implemented Epoxy Resin / PTFE
production

Expanded the GregTech API: There is now a convenience method for
automatically creating polymerization recipes.

Disabled recipes for the Salty Water Potion in favor of a new Salt Water
fluid.
Slightly nerfed Sulfuric Acid production from Sulfur and Water to make
them align with other acids and bases (1 Dust / 1000L Chemical + 1000L
Water -> 1000L acid/base)

Commit:
Added additional recipes for acids and bases

Commit:
Tweaked Titanium, implemented Silicone and Gunpowder

Carbon Monoxide can now be obtained as an optional byproduct from
Titaniumtetrachloride production.

Added a missing chemical reactor recipe for Gunpowder

sergeygta4pony commented:
Если вам не сложно ответить . Планируется ли введение GregTech - 5,6 ракетного топлива из компонентов 1,1-Диметилгидразин( Гептил ) и Тетраоксид диазота ( Амил ) . В описании содержание компонентов не слишком и сложные . Это очень важно при добавлении к GregTech , mod galacticraft .
Спс за понимание .
If it's not difficult for you to answer. Is the introduction of GregTech - 5.6 propellant from the components 1,1-Dimethylhydrazine (Heptyl) and Diazotetrafluoride (Amil) planned? In the description, the content of the components is not too complex. This is very important when adding to GregTech, mod galacticraft.
ATP for understanding.

JohannesGaessler commented:
I'll need to look at the actual production process first, but I'll consider it.
If possible, please provide me with the systematic name of the propellant.

sergeygta4pony commented:
Если возможно, пожалуйста, предоставить мне с систематическое название вытеснителя.????

Опишите пожалуйста еще раз вопрос но более подробно , Я не понял его после перевода .

Commit:
Made the centrifuge recipes for ash more realistic

Commit:
Expanded Charcoal Byproducts, added recipes that produce them

JohannesGaessler commented:
There are many rocket fuels. I want to know which one you want me to implement. I'm assuming you want me to implement the one that is used on the Russian Proton Rocket. It is a mixture of Dimethylhydrazine (C2H8N2) and Dinitrogen Tetroxide (N2O4). The chemicals you said did not translate well. What I wanted to know is the exact name of the fuel you mean so I can be sure.

sergeygta4pony commented:
Если честно сказать какие компоненты будут именно реализованы в ракетном топливе , это решать вам . Но то что сейчас есть в mod galacticraft . Это не то что хотелось иметь в сборке mod GT - 5/6 .Ведь вы же понимаете что можно разделить ракетные двигатели 1-го и 2-го , 3-го уровня на компоненты более сложных в производстве .Для ракет 1 и 2 уровня с применением Керосина и Кмслорода .3 уровня что то например из применявшихся в США . А для грузовой ракеты самое сложное много компонентное . Например 1,1-Диметилгидразин ( НДМГ ) и Тетраоксид диазота ( Амил ) .А так решать вам .
И спасибо что отвечаете.

draknyte1 commented:
Any idea wtf he said in Russian?
We really need to enforce English in here so that everyone can actually read what's being said or suggested..

Techlone commented:
He suggest three different kinds of rocket fuel for galacticraft's rocket tiers.
For T1 and T2 kerosene like as fuel and oxygen as oxidizer. For T3 something that using in USA.
For a cargo rocket (i don't know, is it exist in galacticraft?) something complex. For example [dimethylhydrazine](https://en.wikipedia.org/wiki/Unsymmetrical_dimethylhydrazine) and [nitrogen tetroxide](https://en.wikipedia.org/wiki/Dinitrogen_tetroxide)

@sergeygta4pony Also what is your native language? I'm sure that is not Russian.

JohannesGaessler commented:
I'm not going to tinker with the internal logic of Galacticraft. In Galacticraft there is only one type of rocket fuel that is used by all rockets and I'm not going to change that. My current plan is to change the currently already existing GT recipe for rocket fuel.

Dream-Master commented:
@JohannesGaessler i like the idea to have different fuels in gc. Why not try to add it or/and ask GC devs to add a config for custom fuels?

JohannesGaessler commented:
I've taken a quick look at the Galacticraft forum. Apparently adding more types of fuel is a WIP.  I will keep an eye on how things turn out but I will definitely not try to implement a similar system myself.

Dream-Master commented:
I think if it is wip it will be for 1.8 and newer because in 1.7.10 i see only bug fixes are resolved in the last versions.

Techlone commented:
I guess new fuels for GC it's not a mainstream of the PR and they might be added later.

Bogdan-G commented:
@Techlone 
>Also what is your native language? I'm sure that is not Russian.

не факт / not fact , this text easy many times translate with google translate (and etc soft), without many drop meaning. Yes it is a clumsy text, the translator chews it better. And we do not write literary language here.

codewarrior0 commented:
> I'm not going to tinker with the internal logic of Galacticraft.

You don't really need to. Galacticraft recognizes several fluids as fuels using a kind of fuzzy match:

https://github.com/micdoodle8/Galacticraft/blob/6f2beed2895d5c0a372a7c354323be3c215f17e3/src/main/java/micdoodle8/mods/galacticraft/core/util/FluidUtil.java#L57-L75

So adding new rocket fuel is as simple as naming your fluid "gregtech rocket fuel" or the like.

JohannesGaessler commented:
@codewarrior0 
What I meant is that I won't add several tiers of rocket fuel because Galacticraft does not seem to support that.

Blood-Asp commented:
This PR is slowly getting so big, i likely need some quick overview about the changes once everything is done. Even now looking through the code changes is not easy to see what exactly all was changed.

JohannesGaessler commented:
Sure thing, I'll need to write patch notes anyways.

Commit:
Implemented Nitric Acid, Rocket Fuel

Commit:
Fixed a sneaky and game-breaking bug that caused quite a lot of fluids
to no longer have recipes for filling cells.

The reason for the bug was that the bit flag 256 is already in use to
designate something as empty.
Since only bit flags up to 128 were documented I had unwittingly used
the 256 bit flag for fluid creation.
The documentation now informs about the function of 256.
Automatic fluid and gas creation are now controlled via booleans.

Big thanks to Techlone for helping me debug.

Dream-Master commented:
@JohannesGaessler are you planing to add Iron(III) chloride. Can be usefull for printed circuit boards (pbc)
I have a own fork from gt and if you not planing to add it i will do in my fork.

JohannesGaessler commented:
I have no plans for Iron Chloride.
Content wise my focus will be on new recipes for materials that already exist.
Specifically I'll work on recipes for nitric acid and sulfuric acid.

Dream-Master commented:
@JohannesGaessler ok thanks for info.

Commit:
Chem reactors can now output 2 items, implemented recipes for Nitric
Acid,added utility methods to get Material cells or integrated circuits,
it's now possible to declare a temperature for automatically created
fluids / gases

Commit:
New recipes, convenience methods, recipe rebalancings

Implemented new recipes for Gunpowder, Saltpeter, Sulfuric Acid 
Gave Sulfur dust a burn value (1600, the same as coal)
Made all chemical recipes stoichiometrically correct whenever possible
(2 H + 1 O -> 1 H2O instead of 2 H + 1 O -> 3 H2O)
Added a convenience method to Materials that lets you get dust
succinctly
Put all simplified recipe additions in a separate method, the config is
still missing

Commit:
Changed the newly added recipes so that they use the new API

Commit:
methods for pipe creation, new plastic pipes, implemented PVC

Added a new plastic: Polyvinyl Chloride, currently only used for item
pipes

Added Polytetrafluoroethylene Fluid Pipes

Fixed the number of fluid cells for polymerization, Fixed the default
alpha value for the MaterialBuilder class

Commit:
Added recipes for centrifuging Stone Dust

The amount of elements you get from centrifuging is proportional to
their mass percentage in the earths continental crust with a tolerance
of ~5-20%

Commit:
Added config for simple chemical recipes and stone dust centifugation

JohannesGaessler commented:
@Blood-Asp I wrote top-level patch notes for my PRs:
https://ftb.gamepedia.com/GregTech_5_Unofficial/Experimental_Changelog
The patch notes are currently only visible in the source code.

Commit:
Added molecular formulas for the new materials, added alternative
charcoal recipes.

Charcoal can now be made by treating Wood Pulp or Sugar with Sulfuric
Acid (takes very long, doesn't scale well)

Commit:
Materials for existing fluids, ash nerf, plastic production svg

Commit:
Added distillation recipe for Phenol, tweaked the graphic for plastics

Commit:
Added an svg for misc chemical recipes, tweaked several recipes

Commit:
Added a graphic for acids, made the javadoc more descriptive

JohannesGaessler commented:
I consider this branch ready to be merged now.
All that's left is to finish the wiki page.
Also please autorize me to push onto the master branch so I won't have to open a new PR each time I need to tweak a recipe.

@draknyte1 
I did not remove or change any of the old functionality.
I only added two booleans that indicate whether a corresponding fluid/gas should automatically be created for a given material (false by default).
Anything else I added is merely a wrapper to make things more convenient.

Commit:
Small tweak to Rocket Fuel to ensure that production cost > burn value

draknyte1 commented:
One big PR doesn't warrant git collaborator status. Only Blood and Arch do merges anyway, so it will make no difference. Like anyone, you just make more small prs and pester them on discord to merge. Because until releases roll out, merge speed doesn't matter.

Commit:
Tweaked the plastics svg

Commit:
Created a non-functional first draft for the Large chemical Reactor

Commit:
Implemented checkMachine for the LCR, added machine casings

Commit:
Deleted prints used for debugging

Commit:
Reverted an erroneous name change for nitration mixture

Commit:
Implemented checkRecipe

Commit:
Added new textures for the LCM, using them now

nictrace commented:
I think, current way of trinitroglycerol production is not realictic at all...

The theme of my dissertation was "TransEsterification of vegetable oils", and I will to propose new technology:

1. tiny pile of sodium + 1000 mb of methanol (thanks for methanol!) = esterification agent (liquid)
2. 1 bucket of E.A. + 6 buckets of vegetable oil = 1 bucket of glycerol + 6 buckets of BioDiesel (!) - fuel with extremely low pollution rate!
3. 3 b. of glycerol + 1 b. of nitration mixture = 4 b. of TNG.

Also one new way of sulfuric acid utilization!

JohannesGaessler commented:
@nictrace 
I'll keep it in mind but I probably won't touch it in the near future.
Disregarding the addition of other mods you don't have a lot of plants that could reasonably give you oils though.
Also the pollution you get in GT does not seem to be related to the emission of carbon dioxide, but rather the emission of microscopic dust.

bHXDgiuMa5yLHyC8 commented:
I like nictrace's suggestion.

Commit:
Merge branch 'ChemistryUpdate' into LargeChemicalReactor

Conflicts:
	src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java

Commit:
Added crafting recipes for the LCH

Commit:
Added a new recipe map for the LCR, expanded the API to allow for custom
NEI stack positioning.

Refactored the name of the Large Boiler Fake Fuels so they're in line
with other names.
Fixed a bug where the LCM was checking for the wrong casing type.

Commit:
Fixed a bug that caused there to be two identical recipes for TiF4

Commit:
Changed the Layout of the LCM to 3x3x3, tweaked the name of casings and
the descriptions of multiblocks to make them consistent.

Commit:
Corrected several chlorination recipes in terms of stoichiometrics

Commit:
Merge branch 'LargeChemicalReactor' into ChemistryUpdate

Commit:
Moved old IC2 Biogas recipes to the simplified recipes.

Commit:
Bugfix: I had accidentally copied the IC2 Biogas recipes instead of
moving them.

Commit:
Added a GUI for the Large Chemical Reactor

Commit:
Moved the Biomass -> Ethanol recipes to the simplified chemical recipes.

Commit:
Fixed stocheometrically incorrect Chloroform recipe.

Commit:
Fixed a bug where the LCM could have a mMaxProgressTime of zero.

Commit:
Fixed a bug where the Chemical Reactor still only had one output slot in
it's machine inventory, causing the second output to be voided.

Commit:
Added recipes for oil desulfurization with Fluids and Cells swapped.

Commit:
Merge branch 'unstable' into ChemistryUpdate

Commit:
Moved the casing IDs for the Large Chemical Reactor: 14, 15 -> 4, 5

Limited the minimal recipe time for the LCM to 2 ticks to avoid bugs.

Commit:
Merge branch 'ChemistryUpdate' of https://github.com/JohannesGaessler/GT5-Unofficial into ChemistryUpdate

Commit:
Changed the MetaGeneratedItem ID of the Large Chem Reactor: 1177 -> 1200

Technus commented:
@JohannesGaessler  if you can change the meta id of large chem reactor again.
1169 - is what our fork currently uses (1200 was occupied by transformer of some sort)

JohannesGaessler commented:
>1200 was occupied by transformer of some sort

Well, in my opinion this is bad practice on your part because it can be reasonably assumed that GT5U will keep adding MetaTileEntities and slowly fill up the ID space from below.
You should have left more space between your IDs and the IDs of GT5U to prevent collisions, or better yet, you should have put your IDs into a range that is not yet used by any GT-related mod.
However, since it's too late for that now, I'll change my ID. Again.

Anyways, if you're going to use IDs that fall into the ID range the standalone mod, please make an entry in GregTech_API, because as of right now it reads:
>1000 -  2047 are used by GregTech.

Commit:
Changed the ID of the LCM again because of GTNH collision: 1200 -> 1169

nictrace commented:
**JohannesGaessler**, thank you for implimentation of my recpies! Also you can use Ethanol instead of Methanol for transesterification without reduction of quality of Biofuel.

One more perspective source of vegetable oils is a wood (tall oil). But it has to be cleared of rosin. We need rosin? :)

Commit:
Fixed Chloromethane and Chloroform recipes missing empty cell outputs

Commit:
Fixed Allyl Chloride recipe consuming a Fluid Cell

Commit:
Made it so that Pyrolyse Oven recipes are consistent

Moved Pyrolyse Oven recipes with Charcoal Byproducts, Wood Gas, Wood
Vinegar, or Wood Tar to OreProcessing to ensure that all log types can
be used.

Commit:
Merge remote-tracking branch 'origin/unstable' into ChemistryUpdate

Conflicts:
	src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_Cleanroom.java
	src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_LargeBoiler.java
	src/main/java/gregtech/loaders/load/GT_FuelLoader.java
	src/main/java/gregtech/loaders/preload/GT_Loader_MetaTileEntities.java
	src/main/java/gregtech/nei/GT_NEI_DefaultHandler.java

Rongmario commented:
Does the rocket fuel work with Advanced Rocketry? We need to start integrate with AR instead of Galacticraft.

JohannesGaessler commented:
I have not put forth any effort to make GT Rocket Fuel compatible with Advanced Rocketry.
From what I can tell for Galacticraft it is enough to have your fluid named Rocket Fuel.

However, i did not add Rocket Fuel, I merely changed the recipe.
It's possible that someone else took care of the compatibility.

Rongmario commented:
Ah, my bad, thought it was something separate.
